src/test/java/tel/schich/javacan/test/CANDumpConverter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * The MIT License
+ * Copyright © 2018 Phillip Schichtel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package tel.schich.javacan.test;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The CANDumpConverter converts a human readable CAN dump into a canplayer replayable CAN dump.
+ * <p>
+ * The input lines are transformed from
+ *
+ * <pre>
+ * " (2019-12-11 17:26:46.545849)  can0  301   [8]  08 4C 01 4E 02 C2 0C 5C"
+ * </pre>
+ *
+ * to
+ *
+ * <pre>
+ * "(1576495052.545849) can0 301#084C014E02C20C5C"
+ * </pre>
+ * <p>
+ * This dump can be replayed with the {@code canplayer} from the {@code can-utils} package.
+ *
+ * @author Maik Scheibler
+ */
+public class CANDumpConverter {
```

Comment:
nitpick: In other places CAN in classnames is written as Can if I'm not mistaken.

Reply:
oops. This class was unintentionally commited, as it serves just for a small use-case where i got a CAN-dump from a client that had the wrong format to be replayable.

If you like it, i will remove this class again. Or if you consider it useful i rename it and it stays.
What shall it be?

Reply:
What is the source of that dump? is that some proprietary tool that created it? Not sure how useful it is, but generally why not. However I'd move it into main in some kind of "tools" or "contrib" package or so.

Reply:
The file was created by calling `candump -t A vcan0 > candump.txt` from the [can-utils package](https://packages.debian.org/de/sid/can-utils) A better choice to get replayable dumps is the `-l` option. But the dump was created to be analyzed by an engineer, so the more human readable form.
This class converts it back to the canplayer` readable format.